packages/dd-trace/test/appsec/iast/security-controls/index.spec.js

```diff
@@ -0,0 +1,356 @@
+const { CUSTOM_SECURE_MARK, COMMAND_INJECTION_MARK } =
+  require('../../../../src/appsec/iast/taint-tracking/secure-marks')
+const { saveIastContext } = require('../../../../src/appsec/iast/iast-context')
+
+const moduleLoadEndChannel = dc.channel('dd-trace:moduleLoadEnd')
+
+const CUSTOM_COMMAND_INJECTION_MARK = CUSTOM_SECURE_MARK | COMMAND_INJECTION_MARK
+
+describe('IAST Security Controls', () => {
+  let securityControls, addSecureMark, iastContext
+
+  describe('configure', () => {
+    let controls, parse, startChSubscribe, endChSubscribe
+
+    beforeEach(() => {
+      controls = new Map()
+      parse = sinon.stub().returns(controls)
+      startChSubscribe = sinon.stub()
+      endChSubscribe = sinon.stub()
+
+      const channels = {
+        'dd-trace:moduleLoadStart': {
+          subscribe: startChSubscribe
+        },
+        'dd-trace:moduleLoadEnd': {
+          subscribe: endChSubscribe
+        }
+      }
+
+      securityControls = proxyquire('../../../../src/appsec/iast/security-controls', {
+        'dc-polyfill': {
+          channel: name => channels[name]
+        },
+        './parser': {
+          parse
+        }
+      })
+    })
+
+    afterEach(() => {
+      securityControls.disable()
+    })
+
+    it('should call parse and subscribe to moduleLoad channels', () => {
+      controls.set('sanitizer.js', {})
+
+      const securityControlsConfiguration = 'SANITIZER:CODE_INJECTION:sanitizer.js:sanitize'
+      securityControls.configure({ securityControlsConfiguration })
+
+      sinon.assert.calledWithExactly(parse, securityControlsConfiguration)
+
+      sinon.assert.calledOnce(startChSubscribe)
+      sinon.assert.calledOnce(endChSubscribe)
+    })
+
+    it('should call parse and not subscribe to moduleLoad channels', () => {
+      const securityControlsConfiguration = 'invalid_config'
+      securityControls.configure({ securityControlsConfiguration })
+
+      sinon.assert.calledWithExactly(parse, securityControlsConfiguration)
+
+      sinon.assert.notCalled(startChSubscribe)
+      sinon.assert.notCalled(endChSubscribe)
+    })
+  })
+
+  describe('hooks', () => {
+    beforeEach(() => {
+      addSecureMark = sinon.stub().callsFake((iastContext, input) => input)
+
+      iastContext = {}
+      const context = {}
+
+      securityControls = proxyquire('../../../../src/appsec/iast/security-controls', {
+        '../taint-tracking/operations': {
+          addSecureMark
+        },
+        '../../../../../datadog-core': {
+          storage: {
+            getStore: sinon.stub().returns(context)
+          }
+        }
+      })
+
+      saveIastContext(context, {}, iastContext)
+    })
+
+    afterEach(() => {
+      securityControls.disable()
+      sinon.restore()
+    })
+
+    function requireAndPublish (moduleName) {
+      const filename = require.resolve(moduleName)
+      let module = require(moduleName)
+
+      const payload = { filename, module }
+      moduleLoadEndChannel.publish(payload)
```

Comment:
why do we need to publish this manually when it is suposed to do automatically on require?

Reply:
This way seemed to me more simple. There's no need to init the tracer...